sdk/servicebus/Azure.Messaging.ServiceBus/README.md

```diff
@@ -56,6 +56,22 @@ ServiceBusClient client = new ServiceBusClient(connectionString);
 
 To see how to authenticate using Azure.Identity, view this [example](#authenticating-with-azureidentity).
 
+### ASP.NET Core
+
+To inject ServiceBusClient as a dependency in an ASP .NET Core app, you can register the client in the `Startup.ConfigureServices` method in `Startup.cs`: 
+```C#
+public void ConfigureServices(IServiceCollection services)
+{
+    services.AddAzureClients(builder =>
+    {
+        builder.AddServiceBusClient(Configuration["connection_string_key"]);
```

Comment:
@AlexGhiondea Should we use `Configuration.GetConnectionString("connection_string_key")` instead?

Reply:
I think GetConnectionString assumes that the config includes a "ConnectionStrings" setting. We should probably include what the config looks like to go along with the code used.

Reply:
That's correct. We could use a config like this:

```json
{
  "ConnectionStrings": {
    "ServiceBus": "<connection_string>"
  }
}
```

Then the Service Bus client registration line becomes:

```csharp
builder.AddServiceBusClient(Configuration.GetConnectionString("ServiceBus"));
```

Reply:
I agree showing the config json is a good idea. For some reason @davidfowl did not want to use .GetConnectionString and instead wanted the indexer. David, why? 
If a service supports AzureKeyCredential, the DI sample should use it instead of using the connection string. The type allows for rotating keys. I am fine with merging this as-is, but I would prefer if we filed and issue to update the sample later to use AzureKeyCredential.

Reply:
@pakrym, @JoshLove-msft, I noticed the DI extension method for servicebus works only with connection strings and not with AzureKeyCredential. Do we not support rotating the keys at runtime?

Reply:
The "rotating keys" part is on the AzureKeyCredential type. Are you envisioning that the user would create the AzureKeyCredential in their startup and then hold onto it? We could have an extension that creates the client using the KeyCredential, but it wouldn't be useful unless the users had access to the credential (which they wouldn't unless they created it themselves).

Reply:
Service Bus does support AAD, so it might be worth showing that in addition to the connection string config.

Reply:
> but it wouldn't be useful unless the users had access to the credential (which they wouldn't unless they created it themselves).

So, are you saying that if the user creates ServiceBusClient manually, they can rotate keys, but if they use DI they cannot? If that's the case, them it seems like a big hole in the DI story.

Reply:
Yes, currently that is how it would work. In order to support KeyCredential DI I think it would have to either be injectable or retrievable from the Client. This is a general issue across any libraries using KeyCredential and not specific to Service Bus though.

Reply:
I think we need a sample showing how to rotate keys with DI. Otherwise we would be potentially sending users onto a dead end road.

Reply:
Updated the sample to account for some of this feedback. I think the sample for how to rotate should go into the larger document. Thoughts?

Reply:
With respect to key rotation and such, it feels like we're getting too deep for the README if we start including that level of detail.   Perhaps we should just ensure that we have a sample dedicated to that concept and link to that from the README?

Reply:
I agree with Jesse, we should have a more advanced sample of rolling AzureNamedKeyCredential but I don't think it should be in the main readme. 

Do we have examples of rolling `AzureNamedKeyCredential` without DI?

Reply:
> With respect to key rotation and such, it feels like we're getting too deep for the README if we start including that level of detail. Perhaps we should just ensure that we have a sample dedicated to that concept and link to that from the README?


@jsquire, @pakrym, my point was that once we start pushing people toward DI, we need to explain how do do various things that are not very obvious when using DI. For example, changing a ctor-using sample to key rotation is something developers should be able to do on their own (if they understand basics of key rotation). Changing DI sample to key rotation is at least non-trivial if not outright impossible/impractical.

Reply:
>my point was that once we start pushing people toward DI, we need to explain how do do various things that are not very obvious when using DI. 

Agreed.